pyro/distributions/delta.py

```diff
@@ -34,7 +34,7 @@ def __init__(self, v, log_density=0.0, event_dim=0, validate_args=None):
         batch_shape = v.shape[:batch_dim]
         event_shape = v.shape[batch_dim:]
         if isinstance(log_density, numbers.Number):
-            log_density = v.new_empty(batch_shape).fill_(log_density)
+            log_density = torch.full(batch_shape, log_density, dtype=v.dtype, device=v.device)
```

Comment:
Not relevant to the cases that I see here, but I just discovered that `torch.full` doesn't generalize well in JIT when the fill value is scalar.

e.g.

```python
>>> a = torch.tensor([0, 0])

>>> def fn1(x):
        return torch.full(a.size(), x, dtype=a.dtype, device=a.device)

>>> torch.jit.trace(fn1, torch.tensor(1.))(torch.tensor(2.))
tensor([1, 1])

>>> def fn2(x):
        return torch.empty_like(a).fill_(x)

>>> torch.jit.trace(fn1, torch.tensor(1.))(torch.tensor(2.))
tensor([2, 2])
```

Reply:
Interesting, it would be good to have this example in tests/infer/test_jit.py